### PR TITLE
Handle GPT-5 token params by provider

### PR DESCRIPTION
**Pull request by PeterDaveHello**
Use max_completion_tokens for OpenAI GPT-5-family Chat Completions and
keep max_tokens for non-openai or non-GPT-5-family paths.

Observed error on GPT-5.1-family responses:
Unsupported parameter: 'max_tokens' is not supported with this model.
Use 'max_completion_tokens' instead.

GitHub Copilot PR summary:

> This pull request adds logic to correctly select the token parameter (`max_tokens` vs. `max_completion_tokens`) when calling chat completion APIs, specifically handling new requirements for OpenAI's GPT-5 chat models. The change is applied in both the OpenAI and custom API integrations, and is thoroughly tested with new unit tests.
> 
> **Token parameter selection improvements:**
> 
> * Introduced `getChatCompletionsTokenParams` in `openai-token-params.mjs` to return `max_completion_tokens` for OpenAI GPT-5 chat models, and `max_tokens` for all other models and providers. This ensures API compatibility as OpenAI transitions to the new parameter.
> * Updated `generateAnswersWithCustomApi` and `generateAnswersWithChatgptApiCompat` to use `getChatCompletionsTokenParams` instead of hardcoding the token parameter, ensuring consistent parameter handling across custom and OpenAI APIs. [[1]](diffhunk://#diff-ed0979d1c89a77cc0b6c60b865b1f2e4f5217a4931ec8f4056381a283de66b02R13) [[2]](diffhunk://#diff-ed0979d1c89a77cc0b6c60b865b1f2e4f5217a4931ec8f4056381a283de66b02L58-R59) [[3]](diffhunk://#diff-b817f1a71787e1acd00caa2811609b5984139009ad81aa78389a5240e8038adbR9) [[4]](diffhunk://#diff-b817f1a71787e1acd00caa2811609b5984139009ad81aa78389a5240e8038adbL146-R150)
> 
> **API function signature and usage updates:**
> 
> * Modified `generateAnswersWithChatgptApiCompat` and its callers to accept a `provider` argument, enabling correct parameter selection based on the API provider. [[1]](diffhunk://#diff-b817f1a71787e1acd00caa2811609b5984139009ad81aa78389a5240e8038adbR107-R108) [[2]](diffhunk://#diff-b817f1a71787e1acd00caa2811609b5984139009ad81aa78389a5240e8038adbR119)
> 
> **Testing:**
> 
> * Added comprehensive tests for `getChatCompletionsTokenParams` to verify correct behavior for various provider/model combinations, including OpenAI GPT-5 models, lookalike names, and edge cases.
> 

## Summary by CodeRabbit

* **New Features**
  * Improved token-parameter handling for better compatibility across AI providers and models, including explicit support for newer OpenAI chat models and avoidance of conflicting token fields in requests.
* **Tests**
  * Added comprehensive unit tests covering token-parameter selection across model name patterns, providers, casing, and edge cases to ensure reliable behavior.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@codex review

File: src/services/apis/openai-token-params.test.mjs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import test from 'node:test'
+import assert from 'node:assert/strict'
+import { getChatCompletionsTokenParams } from './openai-token-params.mjs'
```
[the remark this answers is not in the document]
Comment (PeterDaveHello) [reply src/services/apis/openai-token-params.test.mjs:3]: Will do it later.

File: src/services/apis/openai-token-params.test.mjs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import test from 'node:test'
```
Comment (PeterDaveHello) [inline src/services/apis/openai-token-params.test.mjs:1]: This file was placed at the wrong place 😓
